arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -195,7 +365,6 @@ void arm_core_mpu_enable(void)
 {
 	/* Enable MPU */
 	SYSMPU->CESR |= SYSMPU_CESR_VLD_MASK;
-
 }
```

Comment:
add __DSB(); here and remove the barriers at the end of nxp_mpu_init(), this way barriers are always carried on when enabling/disabling mpu not only at mpu initialization.

Reply:
Yeap, will do. ISB?